File: Python/max_consecutive_ones.py
from typing import List


class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxi = 0
        count = 0
        for num in nums:
            if num==1:
                count+=1
            else:
                maxi = max(count, maxi)
                count = 0

        return max(count, maxi)

if __name__ == "__main__":
    solution = Solution()
    
    # Sample test cases
    test_cases = [
        [1, 1, 0, 1, 1, 1],
        [1, 0, 1, 1, 0, 1],
        [0, 0, 0],
        [1, 1, 1, 1]
    ]
    
    for i, nums in enumerate(test_cases, 1):
        result = solution.findMaxConsecutiveOnes(nums)
        print(f"Test case {i}: {nums}")
        print(f"Output: {result}")
        print()

File: Java/max_consecutive_ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max=0;
        int count = 0;
        for(int i=0;i<nums.length;i++){
            if (nums[i]==1){
                count+=1;
            }
            else{
                max = Math.max(max, count);
                count = 0;
            }
        }
        return Math.max(max, count);
    }
}

public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
    
    int[][] testCases = {
        {1, 1, 0, 1, 1, 1},
        {1, 0, 1, 1, 0, 1},
        {0, 0, 0},
        {1, 1, 1, 1}
    };
    
    for (int i = 0; i < testCases.length; i++) {
        int result = solution.findMaxConsecutiveOnes(testCases[i]);
        System.out.print("Test case " + (i + 1) + ": [");
        for (int j = 0; j < testCases[i].length; j++) {
            System.out.print(testCases[i][j]);
            if (j < testCases[i].length - 1) System.out.print(", ");
        }
        System.out.println("]");
        System.out.println("Output: " + result);
        System.out.println();
    }
}